autoscheduler/scheduler/views.py

```diff
@@ -1,4 +1,95 @@
-# Once we start adding routes, uncomment this line
-# from django.shortcuts import render
+import json
+from typing import List, Tuple
+from rest_framework.views import APIView
+from rest_framework.response import Response
 
-# Create your views here.
+from scheduler.create_schedules import create_schedules
+from scheduler.utils import UnavailableTime, CourseFilter
+from scraper.management.commands.scrape_courses import convert_meeting_time
+from scraper.serializers import SectionSerializer
+from scraper.models import Section
+
+def _parse_course_filter(courses) -> List[CourseFilter]:
+    """ Parses the given courses to retrieve and converts them to CourseFilter objects
+        to be used in create_schedules
+    """
+
+    output = []
+
+    for course in courses:
+        subject = str(course["subject"])
+        course_num = str(course["courseNum"])
+
+        sections = course["sections"]
+
+        honors = course["honors"]
+        web = course["web"]
+
+        output.append(CourseFilter(subject=subject, course_num=course_num,
+                                   section_nums=sections, honors=honors, web=web))
+    return output
+
+def _parse_unavailable_times(availabilities) -> List[UnavailableTime]:
+    """ Parses the availabilities input and converts them to UnavailableTime objects
+        to be used in create_schedules
+    """
+
+    unavailable_times = []
+    for avail in availabilities:
+        start_time = convert_meeting_time(avail["startTime"])
+        end_time = convert_meeting_time(avail["endTime"])
+        day = int(avail["day"])

```

Comment:
Why shouldn't we just send the day in unavailable times as an int?

Reply:
That is a great point, I'll change it with that assumption